src/cctz.h

```diff
@@ -249,8 +278,77 @@ std::string Format(const std::string& format, const time_point& tp,
 //   "00.x" -> 00.x  // exact
 //
 // Errors are indicated by returning false.
+//
+// Example:
+//   const cctz::TimeZone tz = ...
+//   std::chrono::system_clock::time_point tp;
+//   if (cctz::Parse("%Y-%m-%d", "2015-10-09", tz, &tp)) {
+//     ...
+//   }
+template <typename D>
 bool Parse(const std::string& format, const std::string& input,
-           const TimeZone& tz, time_point* tpp);
+           const TimeZone& tz, time_point<D>* tpp);
+
+}  // namespace cctz
+
+//
+// IMPLEMENTATION DETAILS
+//
+namespace cctz {
+
+namespace internal {
+// Floors tp to a second boundary. Optionally returns subseconds.
+template <typename D, typename Subseconds = std::chrono::nanoseconds>
+inline time_point<seconds64> FloorSeconds(const time_point<D>& tp,
+                                          Subseconds* subseconds = nullptr) {
+  auto sec = std::chrono::time_point_cast<seconds64>(tp);
+  auto sub = tp - sec;
+  if (sub.count() < 0) {
+    sec -= seconds64(1);
+    sub += seconds64(1);
+  }
+  if (subseconds) *subseconds = std::chrono::duration_cast<Subseconds>(sub);
+  return sec;
+}
+// Specialization for when tp is already second aligned.
+template <typename Subseconds = std::chrono::nanoseconds>
+inline time_point<seconds64> FloorSeconds(const time_point<seconds64>& sec,
+                                          Subseconds* subseconds = nullptr) {
+  if (subseconds) *subseconds = {};
+  return sec;
+}
+}  // namespace internal
+
+template <typename D>
+inline Breakdown BreakTime(const time_point<D>& tp, const TimeZone& tz) {
+  Breakdown BreakTime(const time_point<seconds64>&, const TimeZone&);
+  return BreakTime(internal::FloorSeconds(tp), tz);
+}
+
+template <typename D>
+inline std::string Format(const std::string& format, const time_point<D>& tp,
+                          const TimeZone& tz) {
+  std::string Format(const std::string&, const time_point<seconds64>&,
+                     const std::chrono::nanoseconds&, const TimeZone&);
+  std::chrono::nanoseconds ns{};
```

Comment:
Remember that we need to generalize (or at least expand) nanoseconds here and below.